lib/ansible/modules/storage/netapp/sf_account_manager.py

```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/python
+
+# (c) 2017, NetApp, Inc
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+
+module: sf_account_manager
+
+short_description: Manage SolidFire accounts
+extends_documentation_fragment:
+    - netapp.solidfire
+version_added: '2.3'
+author: Sumit Kumar (sumit4@netapp.com)
+description:
+- Create, destroy, or update accounts on SolidFire
+
+options:
+
+    state:
+        description:
+        - Whether the specified account should exist or not.
+        required: true
+        choices: ['present', 'absent']
+
+    name:
+        description:
+        - Unique username for this account. (May be 1 to 64 characters in length).
+        required: true
+
+    new_name:
+        description:
+        - New name for the user account.
+        required: false
+        default: None
+
+    initiator_secret:
+        description:
+        - CHAP secret to use for the initiator. Should be 12-16 characters long and impenetrable.
+        - The CHAP initiator secrets must be unique and cannot be the same as the target CHAP secret.
+        - If not specified, a random secret is created.
+        required: false
+
+    target_secret:
+        description:
+        - CHAP secret to use for the target (mutual CHAP authentication).
+        - Should be 12-16 characters long and impenetrable.
+        - The CHAP target secrets must be unique and cannot be the same as the initiator CHAP secret.
+        - If not specified, a random secret is created.
+        required: false
+
+    attributes:
+        description: List of Name/Value pairs in JSON object format.
+        required: false
+
+    account_id:
+        description:
+        - The ID of the account to manage or update.
+        required: false
+        default: None
+
+    status:
```

Comment:
Agreed. Unfortunately, the SDK documentation does not provide much information either, so I am not sure if this can be an arbitrary string for the administrator's log. I have emailed the respective team and will add to the documentation once I get more information.